lib/realtime_web/channels/realtime_channel.ex

```diff
@@ -586,4 +601,33 @@ defmodule RealtimeWeb.RealtimeChannel do
     Logger.error("Start channel error: " <> error_msg)
     {:error, %{reason: error_msg}}
   end
+
+  defp assign_policies(false, db_conn, sub_topic, access_token, claims, socket) do
+    channel =
+      case ChannelsCache.get_channel_by_name(sub_topic, db_conn) do
+        {:error, :not_found} -> raise "Channel #{sub_topic} hasn't been initialized"
```

Comment:
Why do you want to raise here and not return with an error tuple? 

We want to make sure to respond to the client with some message if the Channel is not found or if they do not have read permissions.

Reply:
my bad, this should not raise and should be an error as we should have proper reasons as you mentioned